idaes/generic_models/properties/core/eos/ceos.py

```diff
@@ -327,14 +408,48 @@ def calculate_scaling_factors(b, pobj):
 
     @staticmethod
     def build_parameters(b):
-        b._cubic_type = b.config.equation_of_state_options["type"]
-        cname = b._cubic_type.name
         param_block = b.parent_block()
-
-        if hasattr(param_block, cname+"_kappa"):
-            # Common components already constructed by previous phase
+        if not (b.is_vapor_phase() or b.is_liquid_phase()):
+            raise PropertyNotSupportedError("{} received unrecognized phase "
+                 "name {}. Cubic equation of state supports only Vap and Liq "
+                 "phases.".format(param_block.name, b))
+        
+        if b.config.equation_of_state_options["type"] not in set(
+                item for item in CubicType):
+            raise ConfigurationError("{} Unrecognized option for equation of "
+            "state type: {}. Must be an instance of CubicType "
+            "Enum.".format(b.name, b.config.equation_of_state_options["type"]))
+            
+        ctype = b.config.equation_of_state_options["type"]
+        b._cubic_type = ctype
+        cname = ctype.name
+        
+        # Check to see if ConfigBlock was created by previous phase
+        if hasattr(param_block,cname+"_eos_options"):
+            ConfigBlock = getattr(param_block,cname+"_eos_options")
+            for key, value in b.config.equation_of_state_options.items():
+                if ConfigBlock[key] != value:
+                    raise ConfigurationError("In {}, different {} equation of "
+                "state options for {} are set in different phases, which is "
+                "not supported.".format(b.name, cname, key))
+            # Once the options have been validated, we don't have anything
+            # left to do
+            mixing_rule_a = ConfigBlock["mixing_rule_a"]
+            mixing_rule_b = ConfigBlock["mixing_rule_b"]
+            b._mixing_rule_a = mixing_rule_a
+            b._mixing_rule_b = mixing_rule_b
             return
-
+        
+        
+        setattr(param_block,cname+"_eos_options", deepcopy(CubicConfig))
+        ConfigBlock = getattr(param_block,cname+"_eos_options")
+        ConfigBlock.set_value(b.config.equation_of_state_options)
+            
+        mixing_rule_a = ConfigBlock["mixing_rule_a"]
+        mixing_rule_b = ConfigBlock["mixing_rule_b"]
+        b._mixing_rule_a = mixing_rule_a
```

Comment:
Minor comment, but do we need to attach the rules to the Block? They are not hard to retrieve from the config block, so it seems to be a waste of memory to do this.

Reply:
I am not sure what you mean there. Mixing rules are phase specific (they get set in the "equation_of_state_options" part of the config). My point is that the data is already stored on the model in the config block, so we should just retrieve the information from there when we need it, rather than storing it in multiple places.

Reply:
Mixing rules are set in a phase-specific way, but, as it stands, if one phase using PR has already created `am`, `bm`, etc., the second phase using PR just skips their creation entirely and uses what the first phase created. Also, I think the way we index components like `a` by phase would also cause issues if we wanted to use a cubic EoS for the vapor phase but an activity coefficient model for the liquid phase with a nonvolatile component. The nonvolatile component doesn't have Tc, Pc, or omega defined, so at present the user would have to assign them dummy values so the vapor phase could create the (unused) `am[Liq]`.